projects/storefrontlib/src/lib/pwa/services/add-to-home-screen.service.ts

```diff
@@ -0,0 +1,64 @@
+import { Injectable } from '@angular/core';
```

Comment:
```
import { Injectable } from '@angular/core';
import { PWAModuleConfig } from '../pwa.module-config';
import { GlobalMessageType } from './../../global-message/models/message.model';
import { GlobalMessageService } from '../../global-message/facade/global-message.service';

@Injectable()
export class AddToHomeScreenService {
  private deferredEvent: any;
  canPrompt = false;

  constructor(
    private config: PWAModuleConfig,
    private globalMessageService: GlobalMessageService
  ) {
    if (this.config.pwa.addToHomeScreen) {
      this.init();
    }
  }

  init() {
    window.addEventListener('beforeinstallprompt', event => {
      event.preventDefault();
      this.deferredEvent = event;
      this.canPrompt = true;
    });

    window.addEventListener('appinstalled', () => {
      this.globalMessageService.add({
        type: GlobalMessageType.MSG_TYPE_CONFIRMATION,
        text: 'SAP Storefront was added to your home screen'
      });

      this.canPrompt = false;
      this.deferredEvent = null;
    });
  }

  firePrompt() {
    if (this.deferredEvent) {
      this.deferredEvent.prompt();
    }
  }
}
```
I rewrite this service. I feel it is more readable.

Reply:
Looks better. Changed to this version, but kept canPrompt as observable.